src/node/nodestate.h

```
@@ -255,64 +255,114 @@ namespace ccf
 
-      auto handler = this->rpc_map->find(actor);
-      if (!handler.has_value())
+      const auto parsed_count =
+        parser.execute(response.data(), response.size());
+      if (parsed_count != response.size())
       {
-        throw std::logic_error("Handler has no value");
+        LOG_FAIL_FMT(
+          "Tried to parse {} response bytes, actually parsed {}",
+          response.size(),
+          parsed_count);
+        return false;
       }
-      auto frontend = handler.value();
 
+      if (processor.received.size() != 1)
+      {
+        LOG_FAIL_FMT(
+          "Expected single message, found {}", processor.received.size());
+        return false;
+      }
+
+      const auto body =
+        jsonrpc::unpack(processor.received.front().body, jsonrpc::Pack::Text);
+      const auto result_it = body.find(jsonrpc::RESULT);
+      if (result_it == body.end())
+      {
+        LOG_FAIL_FMT("Create response is error: {}", body.dump());
+        return false;
+      }
+
+      return *result_it;
+    }
+
+    bool send_create_request(const std::vector<uint8_t>& packed)
+    {
       const enclave::SessionContext node_session(
         enclave::InvalidSessionId, node_cert);
       auto ctx = enclave::make_rpc_context(node_session, packed);
 
       ctx->is_create_request = true;
-      ctx->actor = actor;
 
-      frontend->process(ctx);
+      const auto actor_opt = http::extract_actor(*ctx);
+      if (!actor_opt.has_value())
+      {
+        throw std::logic_error("Unable to get actor for create request");
+      }
+
+      const auto actor = rpc_map->resolve(actor_opt.value());
+      auto handler = this->rpc_map->find(actor);
+      if (!handler.has_value())
+      {
+        throw std::logic_error("Handler has no value");
+      }
+      auto frontend = handler.value();
+
+      const auto response = frontend->process(ctx);
+      if (!response.has_value())
+      {
+        LOG_INFO_FMT("No response received from processing create request");
```

Could we remove this log line? In the case of Raft, it should raise an error (it will when checking the return value of `create_and_send_request()`). In the case of PBFT, it will always go through this which is as expected.